src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -99,6 +103,69 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
             DecorateOperationSets();
         }
 
+        private void UpdateParameterNames()
+        {
+            Dictionary<Schema, Dictionary<HttpMethod, int>> usageCounts = new Dictionary<Schema, Dictionary<HttpMethod, int>>();
+            foreach (var operationGroup in _codeModel.OperationGroups)
+            {
+                foreach (var operation in operationGroup.Operations)
+                {
+                    foreach (var request in operation.Requests)
+                    {
+                        var httpRequest = request.Protocol.Http as HttpRequest;
+                        if (httpRequest is null)
+                            continue;
+
+                        var bodyParam = request.Parameters.FirstOrDefault(p => p.In == ParameterLocation.Body);
+                        if (bodyParam is null)
+                            continue;
+
+                        if (!usageCounts.TryGetValue(bodyParam.Schema, out var counts))
+                        {
+                            counts = new Dictionary<HttpMethod, int>();
+                            usageCounts.Add(bodyParam.Schema, counts);
+                        }
+
+                        if (!counts.TryGetValue(httpRequest.Method, out var count))
+                        {
+                            counts.Add(httpRequest.Method, 0);
+                        }
+
+                        counts[httpRequest.Method]++;
+                    }
+                }
+            }
+
+            foreach (var operationGroup in _codeModel.OperationGroups)
+            {
+                foreach (var operation in operationGroup.Operations)
+                {
+                    foreach (var request in operation.Requests)
+                    {
+                        var httpRequest = request.Protocol.Http as HttpRequest;
+                        if (httpRequest is null)
+                            continue;
+
+                        if (httpRequest.Method != HttpMethod.Patch)
+                            continue;
+
+                        var bodyParam = request.Parameters.FirstOrDefault(p => p.In == ParameterLocation.Body);
+                        if (bodyParam is null)
+                            continue;
+
+                        if (!usageCounts.TryGetValue(bodyParam.Schema, out var counts))
+                            continue;
+
+                        if (counts.Count != 1 || !counts.TryGetValue(httpRequest.Method, out var count) || count != 1)
+                            continue;
+
+                        bodyParam.Schema.Language.Default.Name = $"{operationGroup.Key.LastWordToSingular()}UpdateOptions";
```

Comment:
My biggest concern here is that in quite a lot cases, the key of operation group is not the same as the resource name in RPs. Could we have some trial runs on the RPs like `appservice`, or `sql` and see the result?
Note that they might already have some directive to do the renamings, we might need to remove those first before we generate code for them